src/Main.hs

```diff
@@ -26,10 +26,8 @@ data Shape = Circle Vector Double
 
 -- Finds a,b, and c for a^2*x + b*x + c*x = 0, useful for finding intersections
 roots :: Double -> Double -> Double -> [Double]
-roots a b c = findRootsWithDescriminant (b * b - 4 * a *c) a b
-
-findRootsWithDescriminant :: Double -> Double -> Double -> [Double]
-findRootsWithDescriminant desc a b
-    | desc == 0 = [0.5 * (-b)]
-    | desc > 0 = [0.5 * (-b + sqrt desc), 0.5 * (-b - sqrt desc)]
+roots a b c
+    | descriminant == 0 = [0.5 * (-b)]
+    | descriminant > 0 = [0.5 * (-b + sqrt descriminant), 0.5 * (-b - sqrt descriminant)]
     | otherwise = []
+    where descriminant = b * b - 4 * a * c
```

Comment:
so when you're finding the roots, you're actually always finding 2 or zero aren't you? it's just that some times the two roots are the same number right (it's a while since I've had to think about this maths heh)

to be honest though, I don't think I actually use this anywhere in the C implementation of the raytracer. might be worth sitting down at some point and looking at the functions it actually involves